pkg/pattern/drain/chunk.go

```diff
@@ -66,11 +66,37 @@ func (c Chunk) ForRange(start, end model.Time) []logproto.PatternSample {
 			return c.Samples[i].Timestamp >= end
 		})
 	}
-	return c.Samples[lo:hi]
+	if step == timeResolution {
```

Comment:
Shouldn't we support the case where the step is lower than the ingest resolution ? May be change the step to the ingest resolution in that case.

Reply:
This does support lower resolution steps - I added tests for that. This if statement is just to short-circuit the sample generation below for the happy case where we don't need to change the data.

The step param is defaulted to the ingest resolution in the API layer. IMO it makes the code harder to test if `step < TimeResolution` is checked here too, as all our tests need to be based on `TimeResolution` increments which would need updating if the resolution changes in the future. I'm happy to do that if you think its better that way though.